dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClasspathCheckOption.java

```diff
@@ -179,4 +180,21 @@ static void setRepositories(CommandLine commandLine) throws ParseException {
       throw new ParseException("Invalid URL specified for Maven repositories: " + ex.getMessage());
     }
   }
+
+  /** Returns a set of jar files that holds entry point classes. */
+  static ImmutableSet<Path> parseEntryPointJars(CommandLine commandLine, List<Path> inputClasspath)
+      throws RepositoryException {
+    ImmutableSet.Builder<Path> entryPoints = ImmutableSet.builder();
+    if (commandLine.hasOption("a") || commandLine.hasOption('b')) {
+      // For an artifact list (or a BOM), the first elements in inputClasspath are the artifacts
+      // specified the list, followed by their dependencies.
+      int artifactCount = generateArtifacts(commandLine).size();
+      // For Maven artifact list (or a BOM), entry point classes are ones in the list
+      entryPoints.addAll(inputClasspath.subList(0, artifactCount));
```

Comment:
Just return `ImmutableSet.copyOf(inputClasspath.subList(0, artifactCount))` here, and return `ImmutableSet.copyOf(inputClassPath)` below on line 196. Then you don't need the builder at all.

Reply:
Updated.